xptifw/unit_test/CMakeLists.txt

```diff
@@ -53,16 +54,26 @@ if (TARGET LLVMSupport)
   set(SUPPORT_LIB LLVMSupport)
 endif()
 
-target_link_libraries(XPTIFWUnitTests PRIVATE
-  llvm_gtest
-  llvm_gtest_main
-  xptifw
-  ${SUPPORT_LIB}
-)
+if (DEFINED LLVM_EXTERNAL_XPTIFW_SOURCE_DIR)
+  target_link_libraries(XPTIFWUnitTests PRIVATE
+    llvm_gtest
+    llvm_gtest_main
+    xptifw
+    ${SUPPORT_LIB}
+  )
+else()
+  target_link_libraries(XPTIFWUnitTests PRIVATE
+    gtest
+    gtest_main
+    xptifw
+  )
+endif()
 
 add_test(NAME XPTIFWTests COMMAND XPTIFWUnitTests)
 
-add_custom_target(check-xptifw-unittest COMMAND
-  $<TARGET_PROPERTY:XPTIFWUnitTests,RUNTIME_OUTPUT_DIRECTORY>/XPTIFWUnitTests)
-add_dependencies(check-xptifw-unittest XPTIFWUnitTests)
-add_dependencies(check-xptifw check-xptifw-unittest)
+if (DEFINED LLVM_EXTERNAL_XPTIFW_SOURCE_DIR)
```

Comment:
Why do we need to guard like this?

Reply:
@aelovikov-intel The intent was to always support building XPTI inside and outside LLVM and the guard was put in to enable the various checks allowed by the LLVM build environment.

Reply:
I still don't understand. What I see is that we build the unit tests in both scenarios but don't allow users to actually run them in one configuration. Is it an accurate understanding?

Reply:
Resolved in phone call and changes reflect the concern.